disabled student responses from the student test history. If a user m…

dsnyder-fairway opened the pull request:
…anages to access those responses, disable the dropdown

## student test history
<img width="1371" alt="screen shot 2018-02-20 at 10 37 48 am" src="https://user-images.githubusercontent.com/33040425/36442309-3099c960-162a-11e8-8f47-d3b8b8083535.png">

## IGNORE THE ASSESSMENT NAME. If they did manage to access the assessment (manually modifying URL), disable the dropdown
<img width="893" alt="screen shot 2018-02-20 at 10 37 21 am" src="https://user-images.githubusercontent.com/33040425/36442413-8c48b6a4-162a-11e8-8eca-487fcf913c37.png">

dsnyder-fairway commented:
ignore the assessment names on this pr

dsnyder-fairway commented on webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts:
```diff
@@ -67,33 +66,21 @@ export class ResultsByStudentComponent implements OnInit {
-  }
-
-  get isInterim() {
-    return this.assessmentType != AssessmentType.SUMMATIVE;
```
unrelated changes. Just thought I would clean this up

esotrope commented on webapp/src/main/webapp/src/app/student/results/student-results.component.html:
```diff
@@ -50,11 +50,12 @@ <h3 class="blue mb-md">{{'labels.student.results.title' | translate}}</h3>
           [minimumItemDataYear]="minimumItemDataYear"
           [exams]="examsByTypeAndSubject.get(assessmentType).get(subject)"></student-history-iab-table>
-        <student-history-ica-summitive-table
+        <student-history-ica-summative-table
```
👍

esotrope commented on webapp/src/main/webapp/src/app/student/responses/student-responses.component.html:
```diff
@@ -28,7 +28,8 @@ <h3 class="blue mb-md">{{'labels.student.responses.title' | translate}}</h3>
     <p-column field="assessmentItem.position" header="{{'labels.groups.results.assessment.items.cols.number' | translate}}" sortable="true">
       <ng-template let-row="rowData" let-item="rowData.assessmentItem" pTemplate="body">
-        <datatable-row-expander [datatable]="datatable" [row]="row" [text]="item.position"></datatable-row-expander>
+        <datatable-row-expander *ngIf="assessment.isInterim" [datatable]="datatable" [row]="row" [text]="item.position"></datatable-row-expander>
```
The backend should be fixed to return no item responses for summative exam requests here i think. Basically this page should render nothing if accessed for summative exam IDs

dsnyder-fairway replied:
I'll work that in

dsnyder-fairway replied:
done

esotrope commented on webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts:
```diff
@@ -31,11 +36,16 @@ export class StudentHistoryICASummitiveTableComponent implements OnInit {
   ngOnInit(): void {
-    this.actions = this.createActions();
+    if (this.type === AssessmentType.ICA) {
+      this.actions = this.createIcaActions();
```
`createInterimActions()` no?

dsnyder-fairway replied:
we have an separate `student-history-iab-table` so I think that name may lead to confusion

esotrope replied:
gotcha. hm we probably shouldn't eventually though. would probably lead to less code if the columns go through some provider interface that accepts assessment type.

esotrope commented on webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts:
```diff
@@ -67,33 +66,21 @@ export class ResultsByStudentComponent implements OnInit {
-    return this.assessmentType == AssessmentType.IAB;
-  }
-
-  get isInterim() {
```
👍 👍 yay red

esotrope commented on reporting-service/src/main/resources/application.sql.yml:
```diff
@@ -31,6 +31,7 @@ sql:
       findAllForExam: >-
         ${sql.reporting.snippet.selectFromExamItem}
         WHERE e.id = :exam_id
+        AND a.type_id != '3'
```
shouldn't be a string right - i think you can drop the `''`

dsnyder-fairway replied:
you are right, I removed the `''`

esotrope commented:
unit test would have caught the '3' thing - let's add a unit test to assert the no-summatives behavior.

dsnyder-fairway commented:
added an integration test to verify no results found for summative exam with attach exam items. Verified it returns results if we removed the condition where `type != 3`
